ndsl/debug/tooling.py

```diff
@@ -0,0 +1,44 @@
+import inspect
+from functools import wraps
+from typing import Any, Callable
+
+from ndsl.debug.config import ndsl_debugger
+
+
+def instrument(func) -> Callable:
+    @wraps(func)
+    def wrapper(self, *args: Any, **kwargs: Any):
+        if ndsl_debugger is None:
+            return func(self, *args, **kwargs)
+        savename = func.__qualname__
+        params = inspect.signature(func).parameters
+        data_as_dict = {}
+
+        # Positional
+        positional_count = 0
+        for name, param in params.items():
+            if param.kind in (
+                inspect.Parameter.POSITIONAL_ONLY,
+                inspect.Parameter.POSITIONAL_OR_KEYWORD,
+            ):
+                if positional_count == 0:  # self
+                    positional_count += 1
+                    continue
+                if positional_count < len(args) + 1:
+                    data_as_dict[name] = args[positional_count - 1]
+                    positional_count += 1
+        # Keyword arguments
+        for name, value in kwargs.items():
+            if name in params:
+                data_as_dict[name] = value
+        if ndsl_debugger:
```

Comment:
```suggestion
        if ndsl_debugger is not None:
```
not sure if we have consensus or convention for that. In other repos (especially gt4py and dace) people follow the convention to always compare potential `None` objects with `is (not) None`.

Reply:
Yeah I'd rather do the verbose `var is None`